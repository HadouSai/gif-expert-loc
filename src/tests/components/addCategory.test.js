import "@testing-library/jest-dom";
import { shallow } from "enzyme"
import AddCategory from "../../components/AddCategory";

describe('Pruebas en GiffGridItem', () => {

    const setCategories = jest.fn();
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should renderize AddCategory component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should change data', () => {
        const input = wrapper.find('input');
        const value = 'Hello my world';

        input.simulate('change', { target: { value } });
    })

    test('should not post submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).not.toHaveBeenCalled();
    })

    test('should call setcategories and clean the input', () => {
        const input = wrapper.find('input');
        const value = 'Hello Kurumi';
        // tener cuidado por que si lo tomo con wrapper
        // es justo en ese momento como lo tengo que se evaluara
        // si sucede algo despues no se tomara en cuenta
        input.simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).toHaveBeenCalledTimes(1);
        //expect(setCategories).toHaveBeenCalled(expect.any(function));
        expect(input.prop('value')).toBe('');

    })



})