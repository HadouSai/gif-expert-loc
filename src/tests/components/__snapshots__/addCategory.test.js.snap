// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiffGridItem should renderize AddCategory component 1`] = `
<form
  action=""
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
